sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionManagerSample.ts

```diff
@@ -0,0 +1,155 @@
+import { ServiceBusClient, delay, ReceivedMessageWithLock } from "@azure/service-bus";
+import * as dotenv from "dotenv";
+import { env } from "process";
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+interface SessionMessageHandlers {
+  processInitialize(sessionId: string);
+  processMessage(msg: ReceivedMessageWithLock);
+  processError(err: Error, sessionId: string);
+  processClose(reason: "error" | "idle_timeout", sessionId: string);
+}
```

Comment:
If we had to make the sample simpler, I would remove this interface.

Assuming that the intention of this sample is to just show the building blocks rather than provide an implementation that user can copy/paste and call into, all 4 handlers above can be removed.

- Remove the `processInitialize` and `processClose` handlers. We can replace calls to these with a comment saying that this is the place the user can do set-up/tear-down for the session
- Remove the `processMessage and `processError` handlers. The console.log() we have now can be inlined.

I would also make the assumption that the idle timeout feature is built-into the session receiver i.e pass the timeout value in options to `createSessionReceiver()`

Reply:
The balance here is tough.

There's much more code in this sample than in a normal sample and there's multiple spots where the user would need to fill stuff in. We could remove this but it does make it a bit easier to follow, IMO. 

My vote is still to keep them - I actually didn't have them originally and added them in after trying to just read through the sample and see if it was "easy" to figure out which parts to fill in.

Reply:
I worry people will expect they need to write this interface. Can we just make process* bare functions and call them from the appropriate places? It might be a bit more verbose but simpler usage.

Reply:
@ramya-rao-a , @bterlson  - I think what's in there now should make both of you happy :)

- The functions are at file scope now so we still get some nice defined area for the users to see what they need to fill in.
- No more interface. BAD BAD INTERFACE.

Reply:
I would rename `processInitialize` to `processSessionInitialize` and `processClose` to `processSessionClose`. Otherwise as a user I am 20 lines into the sample and now seeing something that I am not sure why I would need to use.